### PR TITLE
build/use poetry

### PR DESCRIPTION
**Pull request by a5r0n**
- build: use poetry for dependency management
- build(deps): Update python dependencies
- build(deps): Update python dev dependencies
- ci(build): Update Dockerfile to use poetry
- ci(actions): use poetry for testing action

### CODE REVIEW

**Comment by a5r0n** [comment]
BTW, i already contribute code, why i need approval for the workflows to run?

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,26 @@
-FROM python:3.10-alpine
+FROM python:3.9-slim as base
```
Comment (vincentinttsh) [inline Dockerfile:1]: I'm curious why you switched to 3.9 instead of 3.10

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
-    && rm -rf /var/cache/apk/* \
-    && rm -rf /tmp/*
-ENTRYPOINT ["proxybroker"]
+COPY poetry.lock pyproject.toml ./
+
+RUN poetry config virtualenvs.create false && \
+    poetry install --no-interaction --no-ansi --no-dev
+
+COPY proxybroker proxybroker
+VOLUME "/data"
```
Comment (vincentinttsh) [inline Dockerfile:23]: Why need a volume？

File: Dockerfile [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+EXPOSE 8080
```
Comment (vincentinttsh) [inline Dockerfile:24]: I think default port is 8888
Please correct me if i am wrong.

File: Dockerfile (hunk 1 above)
Comment (a5r0n) [reply Dockerfile:1]: the loop argument is deprecated at 3.10, so we are not yet ready 3.10.

**Comment by a5r0n** [comment]
@vincentinttsh u are right. i just copy-paste one of my Dockerfile and forget some changes.

File: Dockerfile (hunk 1 above)
Comment (vincentinttsh) [reply Dockerfile:1]: I think we need to merge this file ASAP to change our docker version work fine.
Maybe create another PR about this and merge ASAP ?

**Comment by vincentinttsh** [comment]
LGTM

**Comment by bluet** [comment]
> BTW, i already contribute code, why i need approval for the workflows to run?

Oops not sure why. Maybe bcoz the previous PR was filled by me. But I think it won't be a problem anymore. 😀

File: Dockerfile (hunk 1 above)
Comment (bluet) [reply Dockerfile:1]: Will this be fix in https://github.com/bluet/proxybroker2/pull/88 ?

**Comment by vincentinttsh** [comment]
CI error in https://github.com/bluet/proxybroker2/runs/7942760388?check_suite_focus=true#step:4:40
install poetry in CI now only have v1.1 or v1.2 (ref: https://github.com/snok/install-poetry/tree/main/scripts)

**Comment by bluet** [comment]
@a5r0n can you do a quick rebase?

**Comment by a5r0n** [comment]
just a sec

**Comment by bluet** [comment]
tests are still failing

**Comment by bluet** [comment]
@vincentinttsh @a5r0n poetry installation process is still failing, with `Version 1.1 does not exist.`
https://github.com/bluet/proxybroker2/runs/7943728006?check_suite_focus=true#step:4:40

**Comment by bluet** [comment]
@vincentinttsh @a5r0n poetry installation process is still failing, with `Version 1.1 does not exist.`
https://github.com/bluet/proxybroker2/runs/7943728006?check_suite_focus=true#step:4:40

**Comment by a5r0n** [comment]
finnaly!

**Comment by bluet** [comment]
hooray!!!

**Comment by bluet** [comment]
Ref #82

**Pull request merged**